src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMParams.java

```diff
@@ -0,0 +1,1179 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class represents the hyper-parameters and parameters for {@link CoverageModelEMAlgorithm}
+ *
+ * Works both as an argument collection as well as a getter/setter.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelEMParams {
+
+    public enum PsiUpdateMode {
+        /**
+         * Target-resolved unexplained variance
+         */
+        PSI_TARGET_RESOLVED,
+
+        /**
+         * Isotropic unexplained variance (i.e. one value for all targets)
+         */
+        PSI_ISOTROPIC
+    }
+
+    public enum WSolverStrategy {
+        /**
+         * Perform the M-step update of bias covariates locally
+         */
+        W_SOLVER_LOCAL,
+
+        /**
+         * Perform the M-step update of bias covariates using Spark
+         */
+        W_SOLVER_SPARK
+    }
+
+    public enum CopyRatioHMMType {
+        /**
+         * Update the copy ratio (or copy number) locally
+         */
+        COPY_RATIO_HMM_LOCAL,
+
+        /**
+         * Update the copy ratio (or copy number) using Spark
+         */
+        COPY_RATIO_HMM_SPARK
+    }
+
+    public enum ComputeNodeCommunicationPolicy {
+        /**
+         * Push new data to the compute nodes via broadcast-hash-join method
+         */
+        BROADCAST_HASH_JOIN,
+
+        /**
+         * Make an RDD from the new data and perform a joint-map with the compute note RDDs
+         */
+        RDD_JOIN
+    }
+
+    public static final double DEFAULT_LOG_LIKELIHOOD_TOL = 1e-5;
```

Comment:
I agree. We may be able to get rid of a number of these parameters. Though, generally speaking, I'd rather expose more than less, with good default values and bold **advanced** disclaimers w/ proper documentation as you suggested. This is the case with sophisticated tools like HaplotypeCaller, StarAligner, etc. Soon enough, we _will_ get strange errors from various users many of which can be resolved by changing a certain advanced parameter. Without exposing them, we will have to create patches for them and/or build custom jars.

`CoverageModelArgumentCollection` looks quite tidy now. It can still be refactored into smaller subsets of arguments. I made an issue for going over each parameter, properly documenting them, and marking the advanced ones as advanced.